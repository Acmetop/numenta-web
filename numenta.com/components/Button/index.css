/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

._disabled:disabled {
  background-color: #999 !important;
}


.default {
  composes: bg-near-white from 'tachyons';
  composes: ba from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: ph4 from 'tachyons';
  composes: pv3 from 'tachyons';
  composes: color-blue from '../../static/assets/css/theme.css';
}

.light {
  composes: bg-white from 'tachyons';
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: pa1 from 'tachyons';
  composes: color-blue from '../../static/assets/css/theme.css';
}

.primary {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: ph4 from 'tachyons';
  composes: pv3 from 'tachyons';
  composes: white from 'tachyons';
  composes: bg-color-blue from '../../static/assets/css/theme.css';
}

.short {
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: f5-ns from 'tachyons';
  composes: h2 from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: pv0 from 'tachyons';
  composes: relative from 'tachyons';
  composes: white from 'tachyons';
  composes: bg-color-blue from '../../static/assets/css/theme.css';
  top: -0.1rem;
}


.button {
  composes: br2 from 'tachyons';
  composes: dim from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: pointer from 'tachyons';
  composes: _disabled;
}
