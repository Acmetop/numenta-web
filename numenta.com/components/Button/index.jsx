import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const Button = ({children, theme, type}) => {
  const classes = [styles.button, styles[theme]]

  return (
    <button className={classNames(...classes)} type={type}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: React.PropTypes.any.isRequired,
  theme: React.PropTypes.oneOf(['default', 'light', 'primary']),
  type: React.PropTypes.oneOf(['button', 'menu', 'reset', 'submit']),
}

Button.defaultProps = {
  theme: 'default',
  type: 'button',
}

export default Button
