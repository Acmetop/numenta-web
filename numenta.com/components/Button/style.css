.default {
  composes: bg-white from 'tachyons';
  composes: ba from 'tachyons';
  composes: f5 from 'tachyons';
  composes: pa3 from 'tachyons';
  composes: color-blue from '../../static/assets/css/theme.css';
}

.light {
  composes: bg-white from 'tachyons';
  composes: bn from 'tachyons';
  composes: f4 from 'tachyons';
  composes: pa1 from 'tachyons';
  composes: color-blue from '../../static/assets/css/theme.css';
}

.primary {
  composes: bn from 'tachyons';
  composes: f5 from 'tachyons';
  composes: pa3 from 'tachyons';
  composes: white from 'tachyons';
  composes: bg-color-blue from '../../static/assets/css/theme.css';
}


.button {
  composes: br2 from 'tachyons';
  composes: dim from 'tachyons';
  composes: flex-auto from 'tachyons';
  composes: ma0 from 'tachyons';
  composes: pointer from 'tachyons';
}
