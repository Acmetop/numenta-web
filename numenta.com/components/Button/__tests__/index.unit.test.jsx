import React from 'react'
import renderer from 'react-test-renderer'

import Button from '../../Button'


describe('<Button /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Button type="submit">
        Submit
      </Button>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
