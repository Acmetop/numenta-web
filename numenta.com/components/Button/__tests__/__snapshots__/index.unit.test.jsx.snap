exports[`Button React component Renders correctly 1`] = `
<button
  className="button default"
  disabled={false}
  onClick={undefined}
  type="submit">
  Submit
</button>
`;
